util/fibers/fiber_socket_base.cc

```diff
@@ -42,8 +42,8 @@ void FiberSocketBase::SetProactor(ProactorBase* p) {
     proactor_ = nullptr;
   }
   proactor_ = p;
-
-  OnSetProactor();
+  if (p)
+    OnSetProactor();
```

Comment:
if `p` can be `nullptr`, do we need to add an `else { OnResetProactor(); }` here?

Reply:
we need to call OnResetProactor when we nullify non-null `proactor_`. at this point `proactor_` is guaranteed to be null.